### PR TITLE
Init content audit module

### PR DESCRIPTION
**Pull request by sswguo**

### CODE REVIEW

File: content-audit/api/src/main/java/org/commonjava/propulsor/content/audit/model/FileEvent.java [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+    private Date timestamp;
+
+    public FileEvent( String eventType )
+    {
+        this.eventId = UUID.randomUUID();
+        this.eventVersion = DEFAULT_VERSION;
+        this.eventType = eventType;
+    }
+
+    public String toJson()
```
[the remark this answers is not in the document]
Comment (sswguo) [reply content-audit/api/src/main/java/org/commonjava/propulsor/content/audit/model/FileEvent.java]: Yeah, agree with you. Let's move this out of the POJOs.

File: content-audit/api/src/main/java/org/commonjava/propulsor/content/audit/model/FileEvent.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+    private String locationKey;
+
+    private String path;
+
+    private String eventType;
+
+    private String requestId;
+
+    private Integer eventVersion;
```
[the remark this answers is not in the document]
Comment (sswguo) [reply content-audit/api/src/main/java/org/commonjava/propulsor/content/audit/model/FileEvent.java]: Sure, that makes more sense.

[the remark this answers is not in the document]
Comment (sswguo) [reply content-audit/api/src/main/java/org/commonjava/propulsor/content/audit/model/FileEvent.java]: Added the annotation [@JsonAppend ](https://www.baeldung.com/jackson-advanced-annotations) for it. 

That needs us to set the value on serializing it, like:
ObjectWriter writer 
  = mapper.writerFor(Bean.class).withAttribute("eventVersion", "1.0");
String jsonString = writer.writeValueAsString(beanInstance);

File: content-audit/api/src/main/java/org/commonjava/propulsor/content/audit/model/FileEvent.java [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+
+    private Map<String, String> extra;
+
+    @JsonFormat( shape = JsonFormat.Shape.STRING, pattern = "yyyy/MM/dd HH:mm:ss a z" )
```
[the remark this answers is not in the document]
Comment (sswguo) [reply content-audit/api/src/main/java/org/commonjava/propulsor/content/audit/model/FileEvent.java]: Looks the default pattern for JsonFormat.Shape.STRING is ISO 8601 as the [doc](https://fasterxml.github.io/jackson-annotations/javadoc/2.6/com/fasterxml/jackson/annotation/JsonFormat.html) mentioned. So removed the pattern here.

File: content-audit/api/src/main/java/org/commonjava/propulsor/content/audit/model/FileEvent.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+import java.util.Map;
+import java.util.UUID;
+
+public class FileEvent
+                implements Serializable
+{
+
+    public static final Integer DEFAULT_VERSION = Integer.valueOf( 1 );
+
+    private UUID eventId;
```
[the remark this answers is not in the document]
Comment (sswguo) [reply content-audit/api/src/main/java/org/commonjava/propulsor/content/audit/model/FileEvent.java]: Done.

File: content-audit/api/src/main/java/org/commonjava/propulsor/content/audit/model/FileEvent.java [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+import com.fasterxml.jackson.databind.annotation.JsonAppend;
+
+import java.io.Serializable;
+import java.util.Date;
+import java.util.Map;
+import java.util.UUID;
+
+@JsonAppend( attrs = { @JsonAppend.Attr( value = "eventVersion" ) } )
+public class FileEvent
+                implements Serializable
```
Comment (sswguo) [inline content-audit/api/src/main/java/org/commonjava/propulsor/content/audit/model/FileEvent.java]: I guess it's no need to implement Serializable here if we use Json way to (de)serializing.

File: content-audit/api/src/main/java/org/commonjava/propulsor/content/audit/model/FileEvent.java [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+import com.fasterxml.jackson.annotation.JsonFormat;
+import com.fasterxml.jackson.databind.annotation.JsonAppend;
+
+import java.io.Serializable;
+import java.util.Date;
+import java.util.Map;
+import java.util.UUID;
+
+@JsonAppend( attrs = { @JsonAppend.Attr( value = "eventVersion" ) } )
```
Comment (jdcasey) [inline content-audit/api/src/main/java/org/commonjava/propulsor/content/audit/model/FileEvent.java]: I think I made a bad decision asking you to change this. If we're going to make decisions about deserializing / using the object, we should probably just have it as a normal field. Then later, I think we can actually use it as a polymorphic type selector if we need to, and refactor into two implementations of the event in order to deserialize two incompatible versions of the class.

Let's switch back to just using a field for the eventVersion, and initialize it to a constant int value of 1.

Comment (sswguo) [reply content-audit/api/src/main/java/org/commonjava/propulsor/content/audit/model/FileEvent.java]: Ack, it's back. :)

**Pull request merged**